nodes/core/storage/50-file.js

```diff
@@ -130,25 +137,75 @@ module.exports = function(RED) {
                             var packet = node.data.shift()
                             node.wstream.write(packet.data, function() {
                                 node.send(packet.msg);
+                                done();
                             });
 
                         } else {
                             // Dynamic filename - write and close the stream
                             var packet = node.data.shift()
                             node.wstream.end(packet.data, function() {
                                 node.send(packet.msg);
+                                delete node.wstream;
+                                delete node.wstreamIno;
+                                done();
                             });
-                            delete node.wstream;
-                            delete node.wstreamIno;
                         }
                     }
                 }
             }
+        }
+
+        function processQ(queue) {
+            var msg = queue[0];
+            processMsg(msg, function() {
+                queue.shift();
+                if (queue.length > 0) {
+                    processQ(queue);
+                }
+            });
+        }
+
+        this.on("input", function(msg) {
+            var msgQueue = node.msgQueue;
+            if (msgQueue.push(msg) > 1) {
+                // pending write exists
+                return;
+            }
+            processQ(msgQueue);
+            if (node.closing) {
+                closeNode();
```

Comment:
A node should not receive an `input` message if the close handling has already started. If you are seeing that happen, let me know. But this means I don't think this block is needed.